system-int-test/test/cmr/system_int_test/search/generics_test.clj

```diff
@@ -71,8 +72,164 @@
+              "prov1"
+              "options[name][ignore-case]=true"
+
+              "testing provider pattern match"
+              (inf/plural concept-type-string)
+              "provider"
+              "PRO*"
+              "options[provider][pattern]=true"))
+
+          (testing "Testing generics concept-id parameter search"
+            (are3 [plural-concept-type-name search-parameter concept-id-parameter options-flag]
+              (let [results (search-request plural-concept-type-name (str search-parameter "=" concept-id-parameter (if options-flag (str "&" options-flag) (str ""))))
+                    status (:status results)
+                    body (:body results)]
+                (is (string/includes? body name) "record not found")
+                (is (= 200 status) "wrong http status"))
+
+              "Test concept-id parameter search"
+              (inf/plural concept-type-string)
+              "concept-id"
+              concept-id
+              nil
+
+              "Test concept_id parameter search"
+              (inf/plural concept-type-string)
+              "concept_id"
+              concept-id
+              nil
+
+              "Test conceptId parameter search"
+              (inf/plural concept-type-string)
+              "conceptId"
+              concept-id
+              nil))
+
+        (testing "Testing parameter of native_id"
+          (are3 [plural-concept-type-name search-parameter native-id-parameter options-flag]
+            (let [results (search-request plural-concept-type-name (str search-parameter "=" native-id-parameter (if options-flag (str "&" options-flag) (str ""))))
+                  status (:status results)
+                  body (:body results)]
+              (is (string/includes? body name) "record not found")
+              (is (= 200 status) "wrong http status"))
+
+            "Test native_id parameter search"
+            (inf/plural concept-type-string)
+            "native_id"
+            native-id
+            nil
+
+            "Test native-id parameter search"
+            (inf/plural concept-type-string)
+            "native-id"
+            native-id
+            nil
+
+            "Search using nativeId, exact case search"
+            (inf/plural concept-type-string)
+            "nativeId"
+            native-id
+            "options[native-id][ignore-case]=false"
+
+            "Search using nativeId, and ignoring case true (default)"
+            (inf/plural concept-type-string)
+            "nativeId"
+            (string/upper-case native-id)
+            "options[native-id][ignore-case]=true"
+
+            "Search using native-id pattern parameter search"
+            (inf/plural concept-type-string)
+            "native-id"
+            (str (string/join "" (drop-last native-id)) "*")
+            "options[native-id][pattern]=true"))
+            ;; test for legacy documents but, some generics do not have a guid-id
+            (if guid-id
```

Comment:
Neither the API doc nor this test is very clear on what the `id` parameter is actually matching against. From reading the code, I can see the `id` is actually optional for generic concepts, this is not intuitive. I think we need to make that clear in the API doc and the test here.